A71-xds-fallback.md

```diff
@@ -127,94 +52,160 @@ to [xDS Protocol Specification][resource-does-not-exist].
 Note that only attempts at reading a resource that is not cached will trigger
 the fallback.
 
-## Changes to the code base
+### Related Proposals: 
+* [A27: xDS-Based Global Load Balancing][A27]
+* [A36: xDS-Enabled Servers][A36]
+* [A40: xDS Configuration Dump via Client Status Discovery Service in gRPC][A40]
+* [A47: xDS Federation][A47]
+* [A57: XdsClient Failure Mode Behavior][A57]
 
-1. Add multi-client support to CSDS
-1. Refactor code using XdsClient to support multiple instances. 
-1. Add support for multiple xDS servers in the bootstrap JSON. Current
-implementation only returns the first entry and ignores the rest. This should
-include the changes within the XdsClient to support fallback within a given
-authority.
+## Proposal
 
+Perform the following changes:
 
-#### Refactor code using XdsClient to support multiple instances.
+1. Change global XdsClient from a single instance to a map of instances keyed
+by channel target.
+1. Change xDS bootstrap code to return the full list of servers for each
+xDS authority from the bootstrap config.
+1. Change XdsClient to support fallback within each xDS authority.
 
-Instead of having a single XdsClient, introduce an XdsClient instance
-per target. 
+### Change global XdsClient from a single instance to per-target instances
 
-#### Add support for multiple xDS servers in the bootstrap JSON
+Current gRPC implementation is using a single XdsClient for all data plane
+targets. For fallback, this would mean that all the channels would switch to
+using the data from the fallback xDS control plane which is not desirable (see
+below).
 
-1. For each target, we need to store an ordered list of channels instead of
-just one channel.
-1. Whenever we lose contact with a given server and there is at least
-one resource requested but not cached or have previously lost contact and
-an uncached resource is requested, we create a channel/stream for the next
-server in the list unless we are at the last one.
-1. Whenever we get back in contact with a given server, if we have
-a channel/stream for a server later in the list, close it.
+Changes XdsClient to per-data plane target would enable gRPC to switch to
+fallback configuration only for channels that need xDS resource that have not
+been downloaded yet.
 
-Currently bootstrap JSON supports multiple xDS servers but semantics are not
-explicitly specified.
+This change would change the channels to get XdsClient for their data plane
+target. It will also require changes to the CSDS service to aggregate configs
+from multiple XdsClients. For gRPC Core, this would entail updating
```

Comment:
Instead of describing the C-core specific change, I suggest saying something like "The CSDS RPC service will be changed to get data from all XdsClient instances."